stock_quant_package_restrict_disposable_reuse/models/stock_move_line.py

```diff
@@ -0,0 +1,66 @@
+# 2024 Copyright ForgeFlow, S.L. (https://www.forgeflow.com)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class StockMoveLine(models.Model):
+    _inherit = "stock.move.line"
+
+    result_package_id = fields.Many2one(
+        domain="['|', '|', ('location_id', '=', False), ('location_id', '=', location_dest_id),"
+        " ('id', '=', package_id), '|', ('package_type_id.restrict_reuse', '=', False),"
+        " ('disposable_used', '=', False)]"
+    )
+    package_id = fields.Many2one(
+        domain="[('location_id', '=', location_id), '|', "
+        "('package_type_id.restrict_reuse', '=', False), ('disposable_used', '=', False)]"
+    )
+
+    @api.constrains("package_id", "result_package_id")
+    def _check_disposable_package_constrain(self):
+        for record in self:
+            self._check_disposable_package(record.package_id)
+            self._check_disposable_package(record.result_package_id)
+
+    def _check_disposable_package(self, package):
+        if (
+            package
+            and package.package_type_id.restrict_reuse
+            and package.package_use == "disposable"
+            and package.disposable_used
+        ):
+            raise ValidationError(
+                _(
+                    "You can't reuse a disposable package that has already been used"
+                    " in a delivery."
+                )
+            )
```

Comment:
this method could check both package and result package. Then in the constrain you only need to make one call: self._check_disposable_package()

Also, in the error message it would be good to add the package number